lib/transport/reusable_data/licence.ex

```diff
@@ -5,41 +5,47 @@ defmodule Transport.ReusableData.Licence do
 
   import TransportWeb.Gettext
 
-  @enforce_keys [:name]
   defstruct [:name]
 
+  use ExConstructor
+
   @type t :: %__MODULE__{
     name: String.t
   }
 
+  @mappings %{
+    "fr-lo" => dgettext("reusable_data", "fr-lo"),
+    "odc-odbl" => dgettext("reusable_data", "odc-odbl"),
+    "other-open" => dgettext("reusable_data", "other-open"),
+    "notspecified" => dgettext("reusable_data", "notspecified")
+  }
+
   @doc """
   Initialises a licence struct with a given code. If the code is within a known
   list of codes, it is localised to the current Gettext locale.
 
-  If it is not, then it returns an error.
-
   ## Examples
 
       iex> Licence.new(%{name: "fr-lo"})
       %Licence{name: "Open Licence"}
 
+      iex> Licence.new(%{"name" => "fr-lo"})
+      %Licence{name: "Open Licence"}
+
       iex> Licence.new(%{name: "Lolbertarian"})
       %Licence{name: nil}
 
+      iex> Licence.new(%{})
+      %Licence{name: nil}
+
   """
   @spec new(map()) :: %__MODULE__{}
   def new(%{} = attrs) do
-    case Map.get(attrs, :name) do
-      "fr-lo" ->
-        %__MODULE__{name: dgettext("reusable_data", "fr-lo")}
-      "odc-odbl" ->
-        %__MODULE__{name: dgettext("reusable_data", "odc-odbl")}
-      "other-open" ->
-        %__MODULE__{name: dgettext("reusable_data", "other-open")}
-      "notspecified" ->
-        %__MODULE__{name: dgettext("reusable_data", "notspecified")}
-      _ ->
-        %__MODULE__{name: nil}
-    end
+    attrs
+      |> super
+      |> Map.get(:name)
+      |> List.wrap
```

Comment:
You could just to 
```
attrs
|> Map.get(:name, [])
```

Reply:
What I want to do here is to 1. take the name, 2. create a list, and 3. create a map. In ```Map.get/3```, third argument is just a fallback.